File: server/src/model/MyState.ts
import { Schema, type, ArraySchema, MapSchema } from "@colyseus/schema"
import { Cell } from "./Cell";
import { Player } from "./Player";
import { Item } from "./Item";

export class MyState extends Schema {
    @type({ map: Player })
    players = new MapSchema<Player>();

    @type({ map: Item})
    items = new MapSchema<Item>();

    @type("boolean")
    started: boolean = false;

    @type("boolean")
    ended: boolean = false;

    @type("int64")
    startTime: number = 0;

    @type("int64")
    endTime: number = 0;
}

File: core/src/main/kotlin/io/github/doorbash/hexy/model/MyState.kt
package io.github.doorbash.hexy.model

import io.colyseus.annotations.SchemaField
import io.colyseus.default
import io.colyseus.serializer.schema.Schema
import io.colyseus.serializer.schema.types.MapSchema

class MyState : Schema() {
    @SchemaField("0/map/ref", Player::class)
    var players = MapSchema(Player::class.java)

    @SchemaField("1/map/ref", Item::class)
    var items = MapSchema(Item::class.java)

    @SchemaField("2/boolean")
    var started = Boolean.default

    @SchemaField("3/boolean")
    var ended = Boolean.default

    @SchemaField("4/int64")
    var startTime = Long.default

    @SchemaField("5/int64")
    var endTime = Long.default
}